bindings/rascal/utils/fps.py

```diff
@@ -87,3 +93,194 @@ def fps(feature_matrix, n_select, starting_index=None,
     return_dict["fps_hausdorff_d2"] = lmin_d2
 
     return return_dict
+
+
+class FPSFilter(BaseIO):
+    """Farther Point Sampling (FPS) to select samples or features in a given feature matrix.
+    Wrapper around the fps function for convenience.
+    Parameters
+    ----------
+    representation : Calculator
+        Representation calculator associated with the kernel
+    Nselect: int
+        number of points to select. if act_on='sample per specie' then it should
+        be a dictionary mapping atom type to the number of samples, e.g.
+        Nselect = {1:200,6:100,8:50}.
+    act_on: string
+        Select how to apply the selection. Can be either of 'sample',
+        'sample per species','feature'.
+    starting_index: int
+        Index used to start the FPS selection with.
+
+    """
+
+    def __init__(self, representation, Nselect, act_on='sample per species',
+                 starting_index=0):
+        self._representation = representation
+        self.Nselect = Nselect
+        self.starting_index = starting_index
+        if act_on in ['sample', 'sample per species', 'feature']:
+            self.act_on = act_on
+        else:
+            raise 'Wrong input: {}'.format(act_on)
+
+        # effectively selected list of indices at the transform step
+        # the indices have been reordered for effiency and compatibility with
+        # the c++ routines
+        self.selected_ids = None
+        # for 'sample' selection
+        self.selected_sample_ids = None
+        # for 'sample per species' selection
+        self.selected_sample_ids_by_sp = None
+        # for feature selection
+        self.selected_feature_ids_global = None
+
+        self.fps_minmax_d2_by_sp = None
+        self.fps_minmax_d2 = None
+
+    def fit(self, managers):
```

Comment:
Remind me again why this is called `fit`? Because it looks a lot more like a selection or sampling function.

Reply:
`fit` comes from the sklearn interface. There all the ML objects that can be applied to a dataset in a pipeline implement a `fit` and `predict`/`transform` function.

Reply:
Well, I'm not familiar with the scikit-learn interface, and I don't know what fraction of new users will be. For those who aren't familiar with sklearn it's quite unintuitive naming -- especially as this isn't doing a fit at all, it's just subsampling the training data. I don't think it makes sense to follow such an unintuitive convention.

Reply:
ok